charts/telemetry-collector/templates/configmap.yaml

```
@@ -0,0 +1,78 @@
+{{- include
+      "accelleran.common.configMap"
+      (fromYaml (include "accelleran.telemetry-collector.configMap.args" .))
+-}}
+
+{{- define "accelleran.telemetry-collector.configMap.args" -}}
+{{- $ := . -}}
+
+top:
+  {{ $ | toYaml | nindent 2 }}
+
+name: {{ include "accelleran.common.fullname" (dict "top" $) }}-config
+
+data:
+  config.yaml: |-
+    internal:
+      kafka:
+        url: "{{ include "accelleran.common.bootstrap.kafka.hostname" (dict "top" .) }}:{{ include "accelleran.common.bootstrap.kafka.port" (dict "top" .) }}"
+      {{- if $.Values.config.influxDB }}
+      influxDB:
+        {{- if $.Values.config.influxDB.url }}
+        url: "{{ $.Values.config.influxDB.url }}"
+        {{ else }}
+        url: "http://{{ $.Release.Name }}-influxdb2 }}"
+        {{- end }}
+        org: "{{ $.Values.config.influxDB.org }}"
+      {{- end }}
+    pipelines:
+      {{- $natsHostname := include "accelleran.common.bootstrap.nats.hostname" (dict "top" .) }}
+      {{- $natsPort := include "accelleran.common.bootstrap.nats.port" (dict "top" .) }}
+      {{- range $pipeline := $.Values.config.pipelines }}
+      - name: "{{ $pipeline.name }}" 
+        component: "{{ $pipeline.component }}"
+        vendor: "{{ $pipeline.vendor }}"
+        enabled: "{{ $pipeline.enabled }}"
+        {{- if $pipeline.consumers }}
+        consumers:
+          {{- if $pipeline.consumers.natsConsumer }}
+            natsConsumer:
+              {{- if $pipeline.consumers.natsConsumer.url }}
+              url: "{{ $pipeline.consumers.natsConsumer.url }}"
+              {{ else }}
+              url: "{{ $natsHostname }}:{{ $natsPort }}"
+              {{- end }}
+              subjects: "{{ $pipeline.consumers.natsConsumer.subjects }}"
+          {{- end }}
+          {{- if $pipeline.consumers.udpConsumer }}
+            udpConsumer:
+              port: "{{ $pipeline.consumers.udpConsumer.port }}"
+          {{- end }}
+        {{- end }}
+        {{- if $pipeline.publishers }}
+        publishers:
+          {{- if $pipeline.publishers.kafkaPublisher }}
+            kafkaPublisher:
+              topic: "{{ $pipeline.publishers.kafkaPublisher.topic }}"
+          {{- end }}
+          {{- if $pipeline.publishers.influxPublisher }}
+            influxPublisher:
+              {{- if $pipeline.publishers.influxPublisher.defaultBucket }}
+              defaultBucket: "{{ $pipeline.publishers.influxPublisher.defaultBucket }}"
+              {{- end }}
+              {{- if $pipeline.publishers.influxPublisher.buckets }}
+              buckets: {
+                {{- range $key, $value := $pipeline.publishers.influxPublisher.buckets }}
+                  "{{ $key }}": "{{ $value }}",
+                {{- end }}
+              }
+              {{- end }}
+          {{- end }}
+        {{- end }}
```

I would again do something like `udpConsumer` so that everything can be injected with future additions in mind that won't need a chart update. Only there are some mandatory things we then maybe want to check still (e.g. the topic if publishing to kafka). For that we can use things like `required` and/or `fail`.

Again more of a sketch of what it could look like as I didn't test this:

```suggestion
        {{- with $pipeline.publishers }}
          {{ if and $pipeline.publishers.kafkaPublisher (not $pipeline.publishers.kafkaPublisher.topic) }}
            {{ fail "When specifying a kafka publisher also a kafka topic needs to be provided" }}
          {{- end }}
        publishers: {{ . | toYaml | nindent 10 }}
        {{- end }}
```

---

Thanks - have updated and put a catch in for influx not having buckets